src/steps/05-PerformanceRequirements/DOW/ComputeForm.spec.ts

```diff
@@ -0,0 +1,700 @@
+
+        // available class levels should be same as 
+        // modal selected options
+        Vue.nextTick(()=>{
+          expect(wrapper.vm.$data.avlClassificationLevelObjects).toEqual(
+            wrapper.vm.$data.modalSelectedOptions
+          )
+        })
+      })
+
+      it("tests if only one option was selected in classification levels modal then" +
+         "checkSingleClassification is to be called", async () => {
+        
+        await wrapper.setData({
+          allClassificationLevels: allClassificationLevels,
+          modalSelectedOptions: ["class3"]
+        })
+
+        //closes modal and sets avlClassificationLevelObjects with 
+        //class levels selected in modal
+        const classLevelsChangedSpy = await jest.spyOn(wrapper.vm, "classificationLevelsChanged");
+        await wrapper.vm.classificationLevelsChanged();
+        expect(classLevelsChangedSpy).toHaveBeenCalled();
+        
+        //expects modal to be closed and data props to be set accordingly
+        expect(wrapper.vm.$data.showDialog).toBe(false);
+        expect(wrapper.vm.$data.avlClassificationLevelObjects.length).toEqual(1);
+
+        const checkSingleClassSpy = await jest.spyOn(wrapper.vm, "checkSingleClassification");
+        await wrapper.vm.checkSingleClassification();
+        expect(checkSingleClassSpy).toHaveBeenCalled();
+      })
+
+      it("tests if multiple classification levels were selected in modal and are not what " + 
+          "was originally selected reset computeForm.classificationLevels", async () => {
+        computeData.classificationLevel = 'class3';
+        await wrapper.setData({
+          allClassificationLevels: allClassificationLevels,
+          modalSelectedOptions: ["class4", "class5"]
+        })
+
+        //closes modal and sets avlClassificationLevelObjects with 
+        //class levels selected in modal
+        const classLevelsChangedSpy = await jest.spyOn(wrapper.vm, "classificationLevelsChanged");
+        await wrapper.vm.classificationLevelsChanged();
+        expect(classLevelsChangedSpy).toHaveBeenCalled();
+
+        // if the classification level that was selected was removed via the modal,
+        // reset computeData.classificationLevel
+        Vue.nextTick(() => {
+          expect(wrapper.vm.$props.computeData.classificationLevel).toBe("");
+        })
+      });
+    });
+
+    describe("Available Classification Levels Radio Group", () => {
+      it("tests if no items are loaded, then classificationRadioOptions array has length===0", async () => {
+        //set `avlClassificationLevelObjects` with data
+        await wrapper.setData({
+          avlClassificationLevelObjects: []
+        })
+        const spy = await jest.spyOn(wrapper.vm, "setAvlClassificationLevels");
+        await wrapper.vm.setAvlClassificationLevels();
+
+        // call `setAvlClassificationLevels` to transform data into 
+        // radio button list
+        expect(spy).toHaveBeenCalled();
```

Comment:
The only time you'd actually want to use `expect(spy).toHaveBeenCalled()` would be when you have function A which calls function B, and you spy on B to make sure it really does get called when you call `A()`. What's happening here is spying on A, calling it and then validating that you called it (which of course you did).

Reply:
I've always done it this way and here is why...
Click here https://jestjs.io/docs/jest-object#jestspyonobject-methodname and then view the Example test code.

![image](https://user-images.githubusercontent.com/84856468/183494093-0b3e0ca9-187a-4c4f-a46c-31c192c52960.png)

But since you have pointed it out as being redundant, that makes sense...

Reply:
Removed all references to spy, as they are not needed b/c we are calling the methods directly.